custom/ewsghana/reports/email_reports.py

```diff
@@ -54,8 +54,8 @@ def _stock_status(status, loc):
 
         locations = self.get_locations(self.config['location_id'], self.config['domain'])
 
-        products = self.unique_products(locations)
-        row_data = {product.name: defaultdict(lambda: 0) for product in products}
+        row_data = defaultdict(lambda: {'total_fac': 0, 'reported_fac': 0,
+                                        'stockout': 0, 'low': 0, 'overstock': 0, 'adequate': 0})
```

Comment:
@gcapalbo The problem was that some locations are reporting products not assigned to them (or maybe they have this product assigned before). 

Regarding to your first question. I changed this a little bit and now dict is initialized with all active products from domain.